### PR TITLE
Refresh Token

### PR DESCRIPTION
**Pull request by Podgore**

### CODE REVIEW

File: StudyHub/StudyHub.BLL/Services/AuthService.cs [hunk 1]
```diff
@@ -83,16 +87,19 @@ public async Task<AuthSuccessDTO> RefreshTokenAsync(string accessToken, string r
...
-            throw new NotFoundException("User with this id does not exist");
-
-        if (DateTimeOffset.UtcNow > user.RefreshToken.ExpiryDate)
+        if (DateTimeOffset.UtcNow > existingToken.ExpiryDate)
             throw new ExpiredException("Refresh token is expired");
 
-        if (user.RefreshToken.Token != refreshToken)
-            throw new IncorrectParametersException("Refresh token is invalid");
+        if (existingToken.Invalidated)
+            throw new TokenValidatorException($"This token is invaludated: {refreshToken}");
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/AuthService.cs:97]: typo

File: StudyHub/StudyHub.BLL/Services/TokenService.cs [hunk 2]
```diff
@@ -15,12 +17,17 @@ public class TokenService : ITokenService
 {
     private readonly JwtSettings _settings;
     private readonly TokenValidationParameters _tokenValidationParametrs;
+    private readonly IRepository<RefreshToken> _refreshTokenRepository;
+    private readonly UserManager<User> _userManager;
 
-    public TokenService(IOptions<JwtSettings> settings, TokenValidationParameters tokenValidationParametrs)
+    public TokenService(IOptions<JwtSettings> settings, TokenValidationParameters tokenValidationParametrs, IRepository<RefreshToken> refreshTokenRepository, UserManager<User> userManager)
```
Comment (aquaprogit) [inline StudyHub/StudyHub.BLL/Services/TokenService.cs:23]: format each param on new line

**Review by aquaprogit — DISMISSED** [review]

File: StudyHub/StudyHub.Common/Requests/InviteUsersRequest.cs [hunk 3]
```diff
@@ -1,6 +1,6 @@
-﻿namespace StudyHub.Common.DTO.UserInvitation;
+﻿namespace StudyHub.Common.Requests;
 
-public class InviteUsersDTO
+public class InviteUsersRequest
```
Comment (fedyafedko) [inline StudyHub/StudyHub.Common/Requests/InviteUsersRequest.cs:3]: Why here request, we mapping it in UserInvitationService?

**Review by fedyafedko — CHANGES_REQUESTED** [review]

File: StudyHub/StudyHub.Common/Requests/InviteUsersRequest.cs (hunk 3 above)
Comment (Podgore) [reply StudyHub/StudyHub.Common/Requests/InviteUsersRequest.cs:3]: Not correct, firstly we have request from the user, than based on what we got creating DTO, and finally mapped it in InvitedUser entity

Comment (aquaprogit) [reply StudyHub/StudyHub.Common/Requests/InviteUsersRequest.cs:3]: > Not correct, firstly we have request from the user, than based on what we got creating DTO, and finally mapped it in InvitedUser entity

+++

**Review by aquaprogit — APPROVED** [review]

**Review by fedyafedko — APPROVED** [review]

**Pull request merged**